### PR TITLE
[FIX] mail: fix placeholder scrollbar with long name

### PR DESCRIPTION
**Pull request by phenix-factory**
Currently, Chrome show a scrollbar when the placeholder is multilined.
Overflow hidden will avoid the scrollbar and we place a nowrap on the
placeholder text to avoid the multiline.

task-2416058

### CODE REVIEW

File: addons/mail/static/src/components/composer_text_input/composer_text_input.scss [hunk 1]
```diff
@@ -20,7 +20,11 @@
     resize: none;
     border-radius: $o-mail-rounded-rectangle-border-radius-lg;
     border: none;
-    overflow: auto;
+    overflow: hidden;
```
Comment (seb-odoo) [inline addons/mail/static/src/components/composer_text_input/composer_text_input.scss:23]: This will remove the vertical scrollbar when typing a long message, which is not good.

Maybe ok if limited to overflow-x

File: addons/mail/static/src/components/composer_text_input/composer_text_input.scss [hunk 2]
```diff
@@ -20,7 +20,11 @@
...
+
+    &:placeholder-shown {
+        white-space: nowrap;
```
Comment (seb-odoo) [inline addons/mail/static/src/components/composer_text_input/composer_text_input.scss:26]: Without ellipsis this looks bugged, but css ellipsis doesn't seem to have effect on placeholder?
Maybe we'll still need to do a js ellipsis on the placeholder value?

File: addons/mail/static/src/components/composer_text_input/composer_text_input.scss (hunk 1 above)
Comment (phenix-factory) [reply addons/mail/static/src/components/composer_text_input/composer_text_input.scss:23]: Works the same with `overflow-x` but it allow the scroll bar when needed, so it's good solution

File: addons/mail/static/src/components/composer_text_input/composer_text_input.scss (hunk 2 above)
Comment (phenix-factory) [reply addons/mail/static/src/components/composer_text_input/composer_text_input.scss:26]: Well I think it's more because the placeholder is on an textarea. You find code online that suggest that ellipsis works on input text.

File: addons/mail/static/src/components/composer_text_input/composer_text_input.scss [hunk 3]
```diff
@@ -20,7 +20,11 @@
     resize: none;
     border-radius: $o-mail-rounded-rectangle-border-radius-lg;
     border: none;
-    overflow: auto;
+    overflow-x: hidden;
+
+    &:placeholder-shown {
+        white-space: nowrap;
+    }
```
Comment (alexkuhn) [inline addons/mail/static/src/components/composer_text_input/composer_text_input.scss:27]: I think what you want is to apply this ruleset instead:

```scss

&::placeholder {
    white-space: nowrap;
    overflow-x: hidden;
    text-overflow: ellipsis;
}
```

Comment (seb-odoo) [reply addons/mail/static/src/components/composer_text_input/composer_text_input.scss:27]: Good find, that might be just what we need.

Comment (phenix-factory) [reply addons/mail/static/src/components/composer_text_input/composer_text_input.scss]: Well actually, this is close to the first solution I tested, but the ellipsis doesn't work.
The result look the same, but with `::placeholder` we can move the overflow. I think it's a better solution then.

Comment (alexkuhn) [reply addons/mail/static/src/components/composer_text_input/composer_text_input.scss]: At least I've tested on Safari and `text-overflow: ellipsis` works as expected with my suggestion.

Comment (alexkuhn) [reply addons/mail/static/src/components/composer_text_input/composer_text_input.scss]: Yeah... For some reasons, only Safari does the desired `text-overflow: ellipsis;`. It doesn't work on Chrome and Firefox...

I don't think we can solve this problem on current versions of Chrome/Firefox with just CSS. I wouldn't attempt solving it with JS code.

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

File: addons/mail/static/src/components/composer_text_input/composer_text_input.scss [hunk 4]
```diff
@@ -20,7 +20,12 @@
     resize: none;
     border-radius: $o-mail-rounded-rectangle-border-radius-lg;
     border: none;
-    overflow: auto;
```
Comment (seb-odoo) [inline addons/mail/static/src/components/composer_text_input/composer_text_input.scss:23]: @robodoo r-

Wait why is this removed? We need the keep the (vertical) scroll bar for textarea

Comment (phenix-factory) [reply addons/mail/static/src/components/composer_text_input/composer_text_input.scss:23]: Reverted

Comment (alexkuhn) [reply addons/mail/static/src/components/composer_text_input/composer_text_input.scss:23]: > @robodoo r-
> 
> Wait why is this removed? We need the keep the (vertical) scroll bar for textarea

I saw this and had the same reaction, but this is pointless for the scrollbar (at least it was on Safari, haven't tried other browsers....)

Comment (seb-odoo) [reply addons/mail/static/src/components/composer_text_input/composer_text_input.scss:23]: By any chance isn't it why there is a scrollbar for the placeholder in chrome?

Comment (alexkuhn) [reply addons/mail/static/src/components/composer_text_input/composer_text_input.scss:23]: Can't reproduce, but it's still good to have this rule. We want scrollbar on content so makes sense to have `overflow: auto`, even if without this rule it puts a scrollbar because side-effects (explicit height is greater than available height from container => scrollbar).

**Comment by alexkuhn** [comment]
@seb-odoo I let you press the "ok" button if all's good now.

File: addons/mail/static/src/components/composer_text_input/composer_text_input.scss (hunk 4 above)
Comment (seb-odoo) [reply addons/mail/static/src/components/composer_text_input/composer_text_input.scss:23]: So after checking I just think `overflow: auto` is actually the default value of textarea so it doesn't matter to keep it or not, and indeed removing it doesn't fix the scroll chrome bug. I guess we'll go with current solution.

**Review by seb-odoo — APPROVED** [review]
@robodoo r+
